web/src/clipboard_handler.ts

```diff
@@ -0,0 +1,36 @@
+import * as blueslip from "./blueslip";
+import {decode_stream_topic_from_url} from "./hash_util";
+import {get_sub_by_id} from "./stream_data";
+
+export function clipboard_handler(link_text: string, after_copy_cb: () => void): void {
+    const stream_topic = decode_stream_topic_from_url(link_text);
+
+    if (!stream_topic) {
+        return;
+    }
+    const stream = get_sub_by_id(stream_topic.stream_id);
+    const stream_name = stream?.name;
+    const topic_name = stream_topic.topic_name;
+    let formatted_url;
+    if (topic_name !== undefined) {
+        formatted_url = `
+        <a href="${link_text}">#${stream_name}>${topic_name}</a>
+       `;
+    } else {
+        formatted_url = `
+        <a href="${link_text}">#${stream_name}</a>
+       `;
+    }
+    const clipboardItem = new ClipboardItem({
+        "text/plain": new Blob([link_text], {
+            type: "text/plain",
+        }),
+        "text/html": new Blob([formatted_url], {type: "text/html"}),
+    });
+    navigator.clipboard
+        .write([clipboardItem])
+        .then(after_copy_cb)
```

Comment:
The `catch` callback will execute even when there is an error in the callback rather than writing to the clipboard.